### ISSUE TITLE
Unable to get data to persist

### ISSUE THREAD

**Issue by cgd1** [jjethwa/rundeck#138]
Hi - firstly, thank you for creating the repo, saved me some time!

What would be the best practice to get the container to retain data/settings after restart?
Any assistance would be greatly appreciated

With respect
Craig

**Comment by jjethwa** [comment]
Hi @cgd1 

Thanks so much!  😄 

I would suggest using volumes to persist your data across restarts and container updates.  At the very minimum you need volumes for /etc/rundeck and /var/lib/mysql.

Optional or config dependent
1.  filesystem method for storing projects needs /var/rundeck.  Note: Rundeck now uses the database for this by default (Configured via RUNDECK_PROJECT_STORAGE_TYPE)
1. filesystem method for key storage needs /var/lib/rundeck/var/storage  (Configured via RUNDECK_STORAGE_PROVIDER)
1. Extra plugins /opt/rundeck-plugins
1. Rundeck logs /var/logs/rundeck

**Comment by cgd1** [comment]
Thank you for the timely response!

Ok, so at run I need to add:
`-v rundeck_data:/etc/rundeck`

Like so:
`docker volume create rundeck_data`
`docker run -p 4440:4440 -t rundeck3:latest -v rundeck_data:/etc/rundeck`

**Comment by jjethwa** [comment]
@cgd1 

That's correct.  You should create separate directories for each volume so maybe something like
```
-v rundeck_etc_rundeck:/etc/rundeck -v rundeck_var_lib_mysql:/var/lib/mysql
```
or similar depending on how you like to organize things 😄  I also like to use exact paths just to make sure nothing weird happens.

**Comment by jjethwa** [comment]
@cgd1 do you see anything getting generated under the volume directories after starting the container?  Are there any errors in the container logs?
```
docker logs <CONTAINER_NAME>
```

**Comment by jjethwa** [comment]
Hi @cgd1 

Any updates?
